scripts/verifyMessages.ps1

```diff
@@ -0,0 +1,66 @@
+#! /usr/bin/env pwsh
+
+# Start timer
+$start_time = Get-Date
+
+# Validate input
+if ($args.Count -ne 2) {
+    Write-Host "Usage: <script> <directory-to-search> <message-declaration-file>"
+    exit 1
+}
+
+$SEARCH_DIR = $args[0]
+$MESSAGE_FILE = $args[1]
+Write-Host "Processing message declarations from $MESSAGE_FILE..."
+
+# Extract message names directly into an array
+$declared_messages = @()
+Get-Content $MESSAGE_FILE | ForEach-Object {
+    if ($_ -match "DECLARE_MESSAGE\(([^,]+),") {
```

Comment:
Can you just read the json file instead of trying to parse the C++ with regex like this?

Reply:
That's a good idea. Originally, I was looping through the `declared_messages` and searching the source code for any instance of the message. (that was a bad idea lol) so I didn't want to deal with the artifact messages. But now that I'm not doing that, I can use the json. :)

Reply:
Never mind. The json files also have artifact messages. There's no way to distinguish messages from CPP world and from artifact world. The only way I can think of that can distinguish (without modifying the json structure) is that the artifact messages are always after the CPP messages. The CPP messages are in alphabetical order so you can guesstimate which is the first artifact message. That said, we don't enforce sorting of messages in the header file.